lib/commands/base.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Commands
+  class Base
+    attr_reader :bot, :chat_id
+
+    def initialize(bot:, chat_id:)
+      @bot = bot
+      @chat_id = chat_id
+    end
+
+    def call
+      raise NotImplementedError, 'You need to follow the interface and define method #call'
+    end
```

Comment:
You can add to this class method
```ruby
def self.call(*args)
  new(*args).call
end
```
and then use commands without `new`
```ruby
command_class.call(bot: bot, chat_id: message.chat.id)
```

Reply:
This solution does not provide a possibility to have an interface and raise an error if it not provided

Reply:
In this solution you only added class method `call` but you don't need to make any changes for object method `call`. It's will be work the same as now

Now:
```ruby
command_class.new(bot: bot, chat_id: message.chat.id).call
# => NotImplementedError
```

After changes:

```ruby
command_class.call(bot: bot, chat_id: message.chat.id)
# => NotImplementedError
```